packages/theme-productForm/src/productForm.js

```diff
@@ -0,0 +1,93 @@
+function serializeArray(form) {
+  let field,
+    l,
+    s = [];
+  if (typeof form === "object" && form.nodeName == "FORM") {
+    const len = form.elements.length;
+    for (let i = 0; i < len; i++) {
+      field = form.elements[i];
+      if (
+        field.name &&
+        !field.disabled &&
+        field.type != "file" &&
+        field.type != "reset" &&
+        field.type != "submit" &&
+        field.type != "button"
+      ) {
+        if (field.type == "select-multiple") {
+          l = form.elements[i].options.length;
+          for (j = 0; j < l; j++) {
+            if (field.options[j].selected)
+              s[s.length] = { name: field.name, value: field.options[j].value };
+          }
+        } else if (
+          (field.type != "checkbox" && field.type != "radio") ||
+          field.checked
+        ) {
+          s[s.length] = { name: field.name, value: field.value };
+        }
+      }
+    }
+  }
+  return s;
+}
+
+class ProductForm {
+  constructor(
+    productForm = "[data-productform]",
+    productJSON = "[data-product-json]",
+    optionInputs = "[data-option-input]"
```

Comment:
Can this be better ? I would like better if we could do an object and then when you initiate it it's looking more like : 

`const Form = new productForm({
     productForm: '.form',
     optionInputs: '.inputs'
})`

rather then guessing in which order your selectors need to be when you `const Form = new productForm('.form', '.inputs', ..)`